doc/introduction/interfaces/tf.rst

```diff
@@ -24,61 +23,78 @@ Tensorflow is imported as follows:
+The :ref:`QNode decorator <intro_vcirc_decorator>` is the recommended way for creating
+:class:`~.QNode` objects in PennyLane. The only change required to construct a TensorFlow-capable
+QNode is to specify the ``interface='tf'`` keyword argument:
 
 .. code-block:: python
 
     dev = qml.device('default.qubit', wires=2)
+
     @qml.qnode(dev, interface='tf')
-    def circuit1(phi, theta):
+    def circuit(phi, theta):
         qml.RX(phi[0], wires=0)
         qml.RY(phi[1], wires=1)
         qml.CNOT(wires=[0, 1])
         qml.PhaseShift(theta, wires=0)
         return qml.expval(qml.PauliZ(0)), qml.expval(qml.Hadamard(1))
 
-The QNode ``circuit1()`` is now a TensorFlow-capable QNode, accepting ``tf.Variable`` and
+The QNode ``circuit()`` is now a TensorFlow-capable QNode, accepting ``tf.Variable`` and
 ``tf.Tensor`` objects as input, and returning ``tf.Tensor`` objects.
 
 >>> phi = tf.Variable([0.5, 0.1])
 >>> theta = tf.Variable(0.2)
->>> circuit1(phi, theta)
+>>> circuit(phi, theta)
 <tf.Tensor: id=22, shape=(2,), dtype=float64, numpy=array([ 0.87758256,  0.68803733])>
 
-Construction from an existing QNode
------------------------------------
-
-Let us first create two basic, NumPy-interfacing QNodes.
+TensorFlow-capable QNodes can also be created using the
+:ref:`QNode class constructor <intro_vcirc_qnode>`:
 
 .. code-block:: python
 
     dev1 = qml.device('default.qubit', wires=2)
-    dev2 = qml.device('forest.wavefunction', wires=2)
+    dev2 = qml.device('default.mixed', wires=2)
 
-    def circuit2(phi, theta):
+    def circuit1(phi, theta):
         qml.RX(phi[0], wires=0)
         qml.RY(phi[1], wires=1)
         qml.CNOT(wires=[0, 1])
         qml.PhaseShift(theta, wires=0)
         return qml.expval(qml.PauliZ(0)), qml.expval(qml.Hadamard(1))
 
-    qnode1 = qml.QNode(circuit2, dev1)
-    qnode2 = qml.QNode(circuit2, dev2)
+    qnode1 = qml.QNode(circuit1, dev1)
+    qnode2 = qml.QNode(circuit1, dev2, interface='tf')
 
-We can convert the default NumPy-interfacing QNodes to TensorFlow-interfacing QNodes by
-using the :meth:`~.QNode.to_tf` method:
+``qnode1()`` is a default NumPy-interfacing QNode, while ``qnode2()`` is a TensorFlow-capable
+QNode:
+
+>>> qnode2(phi, theta)
+<tf.Tensor: id=22, shape=(2,), dtype=float64, numpy=array([ 0.87758256,  0.68803733])>
 
->>> qnode1 = qnode1.to_tf()
+
+.. _tf_interf_convert:
+
+Construction from an existing QNode
+-----------------------------------
+
+Let's say we change our mind and now want ``qnode1()`` to interface with TensorFlow. We can easily
+perform the conversion by using the :meth:`~.QNode.to_tf` method:
+
+>>> qnode1.to_tf()
 >>> qnode1
-<QNode: device='default.qubit', func=circuit, wires=2, interface=TensorFlow>
+<QNode: device='default.mixed', func=circuit1, wires=2, interface=TensorFlow>
+
+``qnode1()`` is now a TensorFlow-capable QNode, as well. Internally, the :meth:`~.QNode.to_tf`
+method uses the :meth:`~.interfaces.tf.TFInterface.apply` function to do the conversion.
```

Comment:
```suggestion
method uses the :meth:`TFInterface.apply <~.interfaces.tf.TFInterface.apply>` function to do the conversion.
```

This was rendering as simply `apply`, so maybe we should try this instead. Neither was linking up to source code, but I'm not good enough at sphinx to figure out why.  Seems like maybe the interfaces are hidden internal objects that don't show up in the documentation.

I still don't think users need this information, especially if the method doesn't even show up in the documentation, but the text should at least be something more informative  than `apply`.

Reply:
Perhaps it could be moved to a call out/hidden dropdown for advanced users?